persistence/test/setup_test.go

```diff
@@ -65,32 +65,37 @@ func NewTestPostgresContext(t *testing.T, height int64) *persistence.PostgresCon
 	ctx, err := testPersistenceMod.NewRWContext(height)
 	require.NoError(t, err)
 
-	db := &persistence.PostgresContext{
-		Height: height,
-		DB:     ctx.(persistence.PostgresContext).DB,
-	}
+	db, ok := ctx.(persistence.PostgresContext)
+	require.True(t, ok)
 
 	t.Cleanup(func() {
 		require.NoError(t, db.Release())
 		require.NoError(t, testPersistenceMod.ResetContext())
 	})
 
-	return db
+	return &db
 }
 
-// REFACTOR: Can we leverage using `NewTestPostgresContext`here by creating a common interface?
 func NewFuzzTestPostgresContext(f *testing.F, height int64) *persistence.PostgresContext {
 	ctx, err := testPersistenceMod.NewRWContext(height)
 	if err != nil {
-		log.Fatalf("Error creating new context: %s", err)
+		fmt.Printf("Error creating new context: %v\n", err)
+		f.FailNow()
 	}
-	db := persistence.PostgresContext{
-		Height: height,
-		DB:     ctx.(persistence.PostgresContext).DB,
+
+	db, ok := ctx.(persistence.PostgresContext)
+	if !ok {
+		fmt.Println("Error casting RW context to Postgres context")
```

Comment:
Needs to follow a pattern.

`log.Fatal()` same for all `fmt.Print...`

Reply:
Done.